tdeh/py-tree
# Feature/directory explorer

## Overview ##
This branch adds the `DirectoryExplorer` class which exposes an `explore()` method that performs a breadth-first search through directories. The invoker can specify whether hidden files and directories are displayed (".[name]") and can set a maximum recursion level.

## How to Test ##
* Run unittests `python -m dir_ascii.tests` or `py.test`
* Set different recursion levels and confirm that explore does not exceed them
* Confirm that hidden files are only returned when the `show_hidden` parameter is set to *True*
